sw/device/tests/alert_handler_lpg_clkoff_test.c

```diff
@@ -0,0 +1,375 @@
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "alert_handler_regs.h"  // Generated.
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static dif_rv_plic_t plic;
+static dif_alert_handler_t alert_handler;
+static dif_clkmgr_t clkmgr;
+static dif_rv_core_ibex_t ibex;
+static const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+
+static plic_isr_ctx_t plic_ctx = {
+    .rv_plic = &plic,
+    .hart_id = kPlicTarget,
+};
+
+// Depends on the clock domain, sometimes alert handler will trigger a spurious
+// alert after the alert timeout. (Issue #2321)
+// So we allow class A interrupt to fire after the real timeout interrupt is
+// triggered.
+static alert_handler_isr_ctx_t alert_handler_ctx = {
+    .alert_handler = &alert_handler,
+    .plic_alert_handler_start_irq_id = kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+    .expected_irq = kDifAlertHandlerIrqClassb,
+    .is_only_irq = false,
+};
+
+/**
+ * Initialize the peripherals used in this test.
+ */
+static void init_peripherals(void) {
+  mmio_region_t base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_plic_init(base_addr, &plic));
+
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR);
+  CHECK_DIF_OK(dif_alert_handler_init(base_addr, &alert_handler));
+
+  CHECK_DIF_OK(dif_clkmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR), &clkmgr));
+
+  mmio_region_t ibex_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_core_ibex_init(ibex_addr, &ibex));
+
+  // Enable all the alert_handler interrupts used in this test.
+  rv_plic_testutils_irq_range_enable(&plic, kPlicTarget,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassd);
+}
+
+/**
+ * Program the alert handler to escalate on alerts upto phase 1 (i.e. wipe
+ * secret) but not trigger reset. Then CPU can check if the correct interrupt
+ * fires and check the local alert cause register.
+ */
+static void alert_handler_config(void) {
+  dif_alert_handler_alert_t alerts[1];
+  dif_alert_handler_class_t alert_classes[1];
+
+  // Enable all incoming alerts and configure them to classa.
+  // This alert should never fire because we do not expect any incoming alerts.
+  //for (int i = 0; i < ALERT_HANDLER_PARAM_N_ALERTS; ++i) {
+  for (int i = 0; i < 1; ++i) {
+    alerts[i] = i+19;
+    alert_classes[i] = kDifAlertHandlerClassA;
+  }
+
+  // Enable alert ping fail local alert and configure that to classb.
+  dif_alert_handler_local_alert_t loc_alerts[] = {
+      kDifAlertHandlerLocalAlertAlertPingFail,
+      kDifAlertHandlerLocalAlertEscalationPingFail,
```

Comment:
When we set `ping_timeout` value to be `2`, the enabled alerts and all escalation pings will timeout.
Escalation ping will always timeout. So I think this might be the reason we see the flaky part.
I would suggest to move the `kDifAlertHandlerLocalAlertEscalationPingFail` to a different class and do not trigger escalation for this local alert.

But if you want a quick check, you can remove this line and see if your test is still flaky :)